pkg/vault/contracts/Authorizer.sol

```diff
@@ -32,258 +32,139 @@ import "./interfaces/IAuthorizer.sol";
 contract Authorizer is IAuthorizer {
     using EnumerableSet for EnumerableSet.AddressSet;
```

Comment:
You're removing usage of EnumerableSet in favor of plain mappings. If we were to do that, we should also get rid of the import and `using` statement.

However, I think we should keep EnumerableSet, especially in this context, as it can be very useful and powerful for automated permission migrations. See for example how we rely on it to [migrate permissions from the BAL token to the BalancerTokenAdmin](https://github.com/balancer-labs/balancer-v2-monorepo/blob/master/pkg/gauges/contracts/BalancerTokenAdmin.sol#L86).

Reply:
Didn't know about the migration. However, the issue is we will need to denormalize the stored data into two or three data structures as before which IMO is unintuitive and error prone. Otherwise, we could make the migration depend on a list of permissions to be granted or revoked. 

But again, didn't know about the migration, if that's a must we can have it back.

Reply:
Hm, now that I give this closer attention I realize that the previous solution wouldn't quite work, as we will now have people with permissions over multiple addresses. So we'd need to keep sets not only of accounts with permission, but also of all of the addresses where at least one account has said permission.

Let's skip this for now and think more about it later. Created https://github.com/balancer-labs/balancer-v2-monorepo/issues/1129